msmart/cli.py

```diff
@@ -25,230 +16,35 @@
-        except KeyboardInterrupt:
-            sys.exit(0)
-
-
-def get_id_from_response(response):
-    if response[64:-16][:6].hex() == '3c3f786d6c20':
-        xml = response[64:-16]
-        root = ET.fromstring(xml.decode(encoding="utf-8", errors="replace"))
-        child = root.find('smartDevice')
-        m = child.attrib
-        return int.from_bytes(bytearray.fromhex(m['devId']), 'little')
-    else:
-        return 0
-
-
-def get_device_info(device_ip, device_port: int):
-    # Create a TCP/IP socket
-    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-    sock.settimeout(8)
-
-    try:
-        # Connect the Device
-        device_address = (device_ip, device_port)
-        sock.connect(device_address)
-
-        # Send data
-        _LOGGER.debug("Sending to {}:{} {}".format(
-            device_ip, device_port, DEVICE_INFO_MSG.hex()))
-        sock.sendall(DEVICE_INFO_MSG)
-
-        # Received data
-        response = sock.recv(512)
-    except socket.error:
-        _LOGGER.info("Couldn't connect with Device {}:{}".format(
-            device_ip, device_port))
-        return bytearray(0)
-    except socket.timeout:
-        _LOGGER.info("Connect the Device {}:{} TimeOut for 8s. don't care about a small amount of this. if many maybe not support".format(
-            device_ip, device_port))
-        return bytearray(0)
-    finally:
-        sock.close()
-    _LOGGER.debug("Received from {}:{} {}".format(
-        device_ip, device_port, response.hex()))
-    return response
-
-
-def support_test(device: scandevice):
-    if device.version == 3:
-        _device = support_testv3(device)
-    else:
-        _device = ac(device.ip, device.id, device.port)
-    if device.type == 'ac':
-        _device.refresh()
-        _LOGGER.debug("{}".format(_device))
-        device.support = _device.support
-
-def support_testv3(device: scandevice):
-    _device = ac(device.ip, device.id, device.port)
-    for udpid in [get_udpid(device.id.to_bytes(6, 'little')), get_udpid(device.id.to_bytes(6, 'big'))]:
-        token, key = gettoken(udpid)
-        auth = _device.authenticate(key, token)
-        if auth:
-            device.token, device.key = token, key
-            return _device
-    return _device
-
-def gettoken(udpid):
-    global Client
-    if Client is None:
-        Client = cloud(Account, Password)
-    if Client.session == {}:
-        Client.login()
-    return Client.gettoken(udpid)
     
-
-def remove_duplicates(device_list: list):
-    newlist = []
-    for i in device_list:
-        if i not in newlist:
-            newlist.append(i)
-    return newlist
-
-
-def id2udpid(data):
-    b = hashlib.sha256(data).digest()
-    b1, b2 = b[:16], b[16:]
-    b3 = bytearray(16)
-    i = 0
-    while i < len(b1):
-        b3[i] = b1[i] ^ b2[i]
-        i += 1
-    return b3
+    try:
+        discovery = MideaDiscovery(account=account, password=password)
+        found_devices = asyncio.run(discovery.get_all())
```

Comment:
`asyncio.run` is a Python 3.7 addition. 
You need to check the version of python in line 10

Reply:
I've changed to support python 3.5